ui/apps/platform/src/Containers/MainPage/Body.tsx

```diff
@@ -83,19 +92,152 @@ const AsyncSystemHealthPagePF = asyncComponent(
     () => import('Containers/SystemHealth/PatternFly/SystemHealthDashboard')
 );
 
+/*
+ * basePath like violationsBasePath = '/main/violations' is key of routeDescriptorMap in routePaths.ts
+ * propPath includes parameters like path="/main/violations/:alertId?"
+ */
+type BaseRouteComponent = {
+    basePath: string;
+    propPath?: string;
+};
+
+/*
+ * Specify featureFlagDependency property of routeDescriptorMap in routePaths.ts
+ * for a brand new route like /main/vulnerability-management/risk-acceptance
+ * for a temporary route like /main/dashboard-pf for superseding conponent
+ */
+type IndependentRouteComponent = {
+    component: ElementType;
+} & BaseRouteComponent;
+
+/*
+ * At transition from development to release:
+ * Delete temporary route like /main/dashboard-pf from routeDescriptorMap in routePaths.ts
+ * Replace pair of independent route components for original and temporary route in routeComponents
+ * with dependent route component with disabled superseded component and enabled superseding component.
+ *
+ * In case of emergency, patch release turns off feature flag which reverts to superseded component.
+ * When feature flag is deleted, replace dependent route component with independent route component.
+ */
+type DependentRouteComponent = {
+    featureFlagDependency: FeatureFlagEnvVar;
+    componentDisabled: ElementType;
```

Comment:
When going through the README the naming of these fields was a little confusing. Even though there wasn't an `is` prefix, they still read like they should be boolean values. What do you think about something like `componentWhenDisabled`, `whenDisabled`, `flagDisabledComponent` or similar?

Reply:
Yes, `componentWhenDisabled` and `componentWhenEnabled` reads more clearly and fluently.